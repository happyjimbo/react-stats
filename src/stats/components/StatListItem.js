import React, {PropTypes} from 'react'
import {ListGroupItem, Label} from 'react-bootstrap'
import './StatListItem.css'

const StatListItem = (props) => {

    const { todayStat, yesterdayStat, weekAverageStat, yesterdayLastWeekStat } = props
    const { statName, click } = props
    const { todayStyle, todayHigherOrLower, todayDiff } = props
    const { lastWeekStyle, lastWeekHigherOrLower, lastWeekDiff } = props

    return (
        <div className="StatListItem">
            <ListGroupItem onClick={e => click(statName)} header={statName} id="group">
                <p id="removeStyle">Remove</p>
                <p id="graphStyle" >Graph â–¼</p>

                <p>Today: <Label bsStyle={todayStyle}>{todayStat}</Label> <Label bsStyle={todayStyle}>{todayDiff}% {todayHigherOrLower} than yesterday</Label></p>

                <p>Yesterday: <Label bsStyle={lastWeekStyle}>{yesterdayStat}</Label> last week: <Label bsStyle="primary">{yesterdayLastWeekStat}</Label> <Label bsStyle={lastWeekStyle}>{lastWeekHigherOrLower}{lastWeekDiff}%</Label></p>
                
                <p>Week average: <Label bsStyle="info">{weekAverageStat}</Label></p>
                
            </ListGroupItem>
        </div>
    )
}

StatListItem.PropTypes = {
    statName: PropTypes.string.isRequired,
    todayStat: PropTypes.string.isRequired,
    yesterdayStat: PropTypes.string.isRequired,
    yesterdayLastWeekStat: PropTypes.string.isRequired,
    weekAverageStat: PropTypes.string.isRequired,        
    click: PropTypes.func.isRequired
}

export default StatListItem