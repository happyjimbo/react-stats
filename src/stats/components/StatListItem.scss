$position : 5px;
$fontSize : 12px;
$red : #dd0000;

.StatListItem {
    margin: auto;
    width: 97%;
    padding: 0;

    #removeStyle {
        position : absolute;
        float: right;
        top: $position;
        right: $position;
        font-size: $fontSize;
        color: $red;

        &:hover {
            text-decoration: underline;
        }
    }

    #graphStyle {
        position : absolute;
        float: right;
        bottom: $position;
        right: $position;
        font-size: $fontSize;
        color: #ccc;

        &:hover {
            text-decoration: underline;
        }
    }

    #error {
        color: $red;
    }
}