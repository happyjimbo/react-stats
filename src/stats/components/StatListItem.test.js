import React from 'react'
import {shallow} from 'enzyme'
import StatListItem from './StatListItem'
import renderer from 'react-test-renderer'

describe('StatListItem', () => {

    const defaultProps = {
        statName: "MyStatName",
        info: "some info",
        selectClick: () => {},
        todayStat: "this is today",
        yesterdayStat: "yesterday, all my troubles seemed so far away",
        yesterdayLastWeekStat: "last week",
        weekAverageStat: "week average"
    }

    const setUp = (extraProps = undefined) => {

        const props = Object.assign({}, defaultProps, extraProps)
        const component = shallow(<StatListItem {...props} />)

        return {
            props,
            component
        }
    }

    it('compares component to snapshot', () => {
        
        const component = renderer.create(
            <StatListItem {...defaultProps} />
        )

        const statListItem = component.toJSON()
        expect(statListItem).toMatchSnapshot()
    })

    it ('runs callback when listGroupItem clicked, setting value to true', () => {

        let clicked = false
        const selectClick = () => { clicked = true; }

        const props = { selectClick }
        const { component } = setUp(props)
        
        expect(clicked).toBe(false)

        const listGroupItem = component.find('ListGroupItem')
        listGroupItem.simulate('click')
        
        expect(clicked).toBe(true)
    })

    it ('displays stats', () => {

        const { component, props } = setUp()

        const today = component.children().contains(props.todayStat)
        expect(today).toBe(true)

        const yesterday = component.children().contains(props.yesterdayStat)
        expect(yesterday).toBe(true)

        const lastweek = component.children().contains(props.yesterdayLastWeekStat)
        expect(lastweek).toBe(true)

        const weekAverage = component.children().contains(props.weekAverageStat)
        expect(weekAverage).toBe(true)
    })
    
})