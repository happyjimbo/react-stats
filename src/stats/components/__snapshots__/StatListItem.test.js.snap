exports[`StatListItem compares component to snapshot 1`] = `
<div
  className="StatListItem">
  <button
    className="list-group-item"
    id="group"
    onClick={[Function]}
    type="button">
    <h4
      className="list-group-item-heading">
      MyStatName
    </h4>
    <p
      className="list-group-item-text">
      <div
        className="" />
      <div />
      <a
        id="removeStyle"
        onClick={[Function]}>
        Remove
      </a>
      <p
        id="graphStyle">
        Graph 
      </p>
      <p>
        Today: 
        <span
          className="label label-default">
          this is today
        </span>
         
        <span
          className="label label-default">
          % 
           than yesterday
        </span>
      </p>
      <p>
        Yesterday: 
        <span
          className="label label-default">
          yesterday, all my troubles seemed so far away
        </span>
         last week: 
        <span
          className="label label-primary">
          last week
        </span>
         
        <span
          className="label label-default">
          %
        </span>
      </p>
      <p>
        Week average: 
        <span
          className="label label-info">
          week average
        </span>
      </p>
    </p>
  </button>
</div>
`;
